src/mysql_protocol.erl

```diff
@@ -736,6 +736,8 @@ encode_param(null) ->
 encode_param(Value) when is_binary(Value) ->
     EncLength = lenenc_int_encode(byte_size(Value)),
     {<<?TYPE_VAR_STRING, 0>>, <<EncLength/binary, Value/binary>>};
+encode_param(Value) when is_list(Value) ->
+    encode_param(unicode:characters_to_binary(Value));
```

Comment:
Thanks for this PR. At first it looked like a simple merge to accept but after looking into it more i have a few concerns...

SQL queries as the query parameter to `mysql:query/2,3,4` are documented as being iodata as opposed to unicode character data.

In `mysql:encode/2`, lists are interpreted as iolists: https://github.com/SergejJurecko/mysql-otp/blob/master/src/mysql_encode.erl#L20 as opposed to unicode character data. I think it is important that the behaviour is identical whether prepared statements are used or encode/2 is used. The return value of mysql:encode/2 is also an iolist that can be used to build queries using simple iolist construction such as `["UPDATE foo SET bar = ", mysql:encode(Conn, Bar), " WHERE id = 1;"]`.

Probably it would make sense to changes all these unicode character data but that would break current behaviour and would mean that we have to do a major version bump if we strictly are to follow semver. Is it worth it? Can we do it with a just minor version bump?

Also (assuming the iodata/unicode issue can be resolved) I'd like a testcase (or a extra line in an existing testcase) to cover `encode_param(Value) when is_list(Value)`.